src/core/components/files-mfs.js

```diff
@@ -32,15 +34,37 @@ module.exports = self => {
     repoOwner: self._options.repoOwner
   })
 
+  const withPreload = fn => (...args) => {
+    const cids = args.reduce((cids, p) => {
+      if (isIpfs.ipfsPath(p)) {
```

Comment:
AFAIK `/ipns`/ paths are not supported by MFS, so HTTP gateway resolves `/ipns/` → `/ipfs/` before passing it to `files.stat` later:

https://github.com/ipfs/js-ipfs/blob/3cade672b1c3d2aa19394e2980cfad9cd94764dc/src/http/gateway/resources/gateway.js#L40-L46

That being said,  is it possible `/ipns/` paths will be supported in the future?

```suggestion
      if (isIpfs.path(p)) {
```

Reply:
I think we'd also have to add the resolve step to this PR if we wanted to "future-proof" the check...right now it would end up sending Peer IDs to the preload nodes...